File: Python/duplicate_encoder.py
# # SOLUTION ONE
# def duplicate_encode(word):
#     mapped = map(lambda c: ")" if word.lower().count(c)
#                  > 1 else "(", list(word.lower()))
#     return "".join(list(mapped))

# SOLUTION TWO
def duplicate_encode(word):
    return "".join([")" if word.lower().count(c) > 1 else "(" for c in word.lower()])


print(duplicate_encode("din"))
print(duplicate_encode("recede"))
print(duplicate_encode("Success"))
print(duplicate_encode("(( @"))

File: JavaScript/previousExercises/duplicateEncoder.js
// // SOLUTION ONE
// function duplicateEncode(word) {
//   const count = (c) => {
//     var count = 0;
//     for (let i = 0; i < arr.length; i++) {
//       if (c === arr[i]) {
//         count++;
//       }
//     }
//     return count > 1 ? ")" : "(";
//   };

//   const arr = word.toLowerCase().split("");
//   const res = arr.map((c) => count(c)).join("");
//   return res;
// }

// SOLUTION TWO
function duplicateEncode(word) {
  return word
    .toLowerCase()
    .split("")
    .map((c, i, w) => {
      return w.indexOf(c) == w.lastIndexOf(c) ? "(" : ")";
    })
    .join("");
}

console.log(duplicateEncode("din"));
console.log(duplicateEncode("recede"));
console.log(duplicateEncode("Success"));
console.log(duplicateEncode("(( @"));
